aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -241,6 +246,52 @@ private List<String> deleteObjectsInBucket(String bucket, Collection<String> obj
     return Lists.newArrayList();
   }
 
+  @Override
+  public Stream<FileInfo> listPrefix(String prefix) {
+    S3URI s3uri = new S3URI(prefix, awsProperties.s3BucketToAccessPointMapping());
+
+    return internalListPrefix(s3uri.bucket(), s3uri.key()).stream()
+        .flatMap(r -> r.contents().stream())
+        .map(o -> new FileInfo(o.key(), o.size(), o.lastModified()));
```

Comment:
I think this needs to reconstruct a URI rather than returning just the key. Otherwise, it isn't really a "location" as we think of them in Iceberg. For example, when listing `"s3://bucket/prefix"`, this might produce `[FileInfo("prefix/file1.parquet", 10, ...), ...]`

Reply:
Good catch, the bucket is actually not returned as part of the object, but I think it can be reconstructed from the original path.

Reply:
Updated and made sure the original scheme was returned.